addon-resizer/nanny/nanny_lib.go

```diff
@@ -74,41 +99,74 @@ type ResourceEstimator interface {
 // PollAPIServer periodically counts the number of nodes, estimates the expected
 // ResourceRequirements, compares them to the actual ResourceRequirements, and
 // updates the deployment with the expected ResourceRequirements if necessary.
-func PollAPIServer(k8s KubernetesClient, est ResourceEstimator, contName string, pollPeriod time.Duration, threshold uint64) {
+func PollAPIServer(k8s KubernetesClient, est ResourceEstimator, pollPeriod, scaleDownDelay, scaleUpDelay time.Duration, threshold uint64) {
+	lastChange := time.Now()
+	lastResult := noChange
+
 	for i := 0; true; i++ {
 		if i != 0 {
 			// Sleep for the poll period.
 			time.Sleep(pollPeriod)
 		}
 
-		// Query the apiserver for the number of nodes.
-		num, err := k8s.CountNodes()
-		if err != nil {
-			log.Error(err)
-			continue
+		if lastResult = updateResources(k8s, est, time.Now(), lastChange, scaleDownDelay, scaleUpDelay, threshold, lastResult); lastResult == override {
+			lastChange = time.Now()
 		}
-		log.V(4).Infof("The number of nodes is %d", num)
+	}
+}
 
-		// Query the apiserver for this pod's information.
-		resources, err := k8s.ContainerResources()
-		if err != nil {
-			log.Errorf("Error while querying apiserver for resources: %v", err)
-			continue
-		}
+// updateResources counts the number of nodes, estimates the expected
+// ResourceRequirements, compares them to the actual ResourceRequirements, and
+// updates the deployment with the expected ResourceRequirements if necessary.
+// It returns true if deployment has been updated.
+func updateResources(k8s KubernetesClient, est ResourceEstimator, now, lastChange time.Time, scaleDownDelay, scaleUpDelay time.Duration, threshold uint64, prevResult updateResult) updateResult {
+	// Query the apiserver for the number of nodes.
+	num, err := k8s.CountNodes()
+	if err != nil {
+		log.Error(err)
+		return noChange
+	}
+	log.V(4).Infof("The number of nodes is %d", num)
+
+	// Query the apiserver for this pod's information.
+	resources, err := k8s.ContainerResources()
+	if err != nil {
+		log.Errorf("Error while querying apiserver for resources: %v", err)
+		return noChange
+	}
 
-		// Get the expected resource limits.
-		expResources := est.scaleWithNodes(num)
+	// Get the expected resource limits.
+	expResources := est.scaleWithNodes(num)
 
-		// If there's a difference, go ahead and set the new values.
-		if !shouldOverwriteResources(int64(threshold), resources.Limits, resources.Requests, expResources.Limits, expResources.Requests) {
-			log.V(4).Infof("Resources are within the expected limits. Actual: %+v Expected: %+v", *resources, *expResources)
-			continue
+	// If there's a difference, go ahead and set the new values.
+	overwrite, op := shouldOverwriteResources(int64(threshold), resources.Limits, resources.Requests, expResources.Limits, expResources.Requests)
```

Comment:
nit: consistency - you're using overwrite in some places and override in other
Can you please unify it?

Reply:
I have changed all occurrences of override to overwrite.